### PR TITLE
Improves status check for installer set and fixes deployment failures 

### PR DESCRIPTION
**Pull request by sm43**
this patch
* Improves status check for installer set and adds tests
* now we delete triggers installerset if ns in deletion, later recreates
* cherry-picks and updates from 0.60.x - Add transformer to fix warnings for PSA
* delete deployment in failed state, & skipping requeuing req for checking status

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]

**Comment by vdemeester** [comment]
@sm43 can you squash your commit ? (aka no merge commit 😝 )

**Comment by sm43** [comment]
> 



> @sm43 can you squash your commit ? (aka no merge commit stuck_out_tongue_closed_eyes )

yeah I was just testing CI 😝 updated now

File: test/e2e/common/05_tektonhubdeployment_test.go [hunk 1]
```diff
@@ -32,7 +32,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func Test_HubDeploymentWithExternalDatabase(t *testing.T) {
+func TestTektonHubDeploymentWithExternalDatabase(t *testing.T) {
+	t.Skip()
```
Comment (sm43) [inline test/e2e/common/05_tektonhubdeployment_test.go:36]: https://github.com/tektoncd/operator/issues/1146

Comment (sm43) [reply test/e2e/common/05_tektonhubdeployment_test.go:36]: need to figure out why

**Pull request merged**